pkg/orchestration/wiring/wiringutil/service_binding.go

```diff
@@ -29,6 +30,20 @@ func ConsumerProducerServiceBinding(consumer, producer voyager.ResourceName, pro
 	return ServiceBinding(bindingResourceName, bindingMetaName, serviceInstanceRef, exposed)
 }
 
+// TODO(kopper): Remove V2 suffix
+func ConsumerProducerServiceBindingV2(consumer, producer voyager.ResourceName, resourceReference wiringplugin.ProtoReference,
+	exposed bool) wiringplugin.WiredSmithResource {
+
+	serviceInstanceRef := smith_v1.Reference{
+		Name:     ReferenceName(resourceReference.Resource, "metadata", "name"),
+		Resource: resourceReference.Resource,
+		Path:     resourceReference.Path,
```

Comment:
Must copy example and modifier fields too!

Reply:
I think we need to have a method on proto reference that takes missing pieces and produces a full smith reference. This will ensure we'll not forget some fields and that if we add new fields we'll get compilation errors in all places where we convert proto to a full ref.

Reply:
```go

// ToReference should be used to augment ProtoReference with missing information to
// get a full Reference.
func (r *ProtoReference) ToReference(name smith_v1.ReferenceName) *smith_v1.Reference {
	return &smith_v1.Reference{
		Name:     name,
		Resource: r.Resource,
		Path:     r.Path,
		Example:  r.Example,
		Modifier: r.Modifier,
	}
}
```
Add this method please.

Reply:
done